pkg/controllers/applicationset_controller.go

```diff
@@ -143,12 +143,12 @@ func (r *ApplicationSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		var message string
 		for _, v := range validateErrors {
 			message = v.Error()
-			break
+			log.Errorf("validation error found during application validation: %s", message)
 		}
 		if len(validateErrors) > 1 {
+			// Only the last message gets added to the appset status, to keep the size reasonable.
```

Comment:
Should we keep the old behavior which adds the first error to appset status?

Reply:
@wtam2018 good question. It's pulling from an [unordered map](https://go.dev/blog/maps#iteration-order), so getting the last message is effectively the same as getting the first message.

Reply:
Maybe I should change the comment to "last encountered message" to clarify?

Reply:
Thanks for the clarification. The comment is fine. It is an unordered map so we are making any claim as to which one of the errors to return.